content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,333 @@
+preparedness.  Some of the recommendations in this checklist may be too
+restrictive or too lax for your specific security needs. Since Kubernetes
+security is not "one size fits all", each category of checklist items should be
+evaluated on their merits.
+{{< /caution >}}
+
+<!-- body -->
+
+## Authentication & Authorization
+
+- [ ] `system:masters` group is not used for user or component authentication after bootstrapping.
+- [ ] The kube-controller-manager is running with `--use-service-account-credentials`
+  enabled.
+- [ ] The root certificate is protected (either an offline CA, or a managed
+  online CA with effective access controls).
+- [ ] Intermediate and leaf certificates have an expiry date no more than 3
+  years in the future.
+- [ ] A process exists for periodic access review, and reviews occur no more
+  than 24 months apart.
+
+See the [Role Based Access Control Good Practices](/docs/concepts/security/rbac-good-practices/)
+for guidance related to authentication and authorization.
+
+After bootstrapping, neither users nor components should authenticate to the
+Kubernetes API as `system:masters`. Similarly, running all of
+kube-controller-manager as `system:masters` should be avoided. In fact,
+`system:masters` should only be used as a break-glass mechanism, as opposed to
+an admin user.
+
+## Network security
+
+- [ ] CNI plugins in-use supports network policies.
+- [ ] Network policies to filter how workloads and services can communicate
+  inside of the cluster are enabled.
+- [ ] Use of a service mesh to encrypt all communications inside of the cluster.
+- [ ] The Kubernetes API is not exposed publicly on Internet.
+- [ ] Access from the workloads to the cloud metadata API is filtered.
+
+A number of [Container Network Interface (CNI) plugins](/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/)
+plugins provide the functionality to
+restrict network resources pods may communicate with. This is most commonly done
+through [Network Policies](/docs/concepts/services-networking/network-policies/)
+which provide a namespaced resource to define rules.
+
+Not all CNI plugins provide encryption in transit. If the chosen plugin lacks this
+feature, an alternative solution could be to use a service mesh to provide that
+functionality.
+
+The `etcd` datastore of the control plane should have controls to limit access.
+Furthermore, mutual TLS (mTLS) should be used to communicate securely with it.
+The certificate authority for this should be unique to `etcd`.
+
+External Internet access to the Kubernetes API server should be restricted
+to not expose the API publicly. You can then use a bastion host to access the server.
+
+If a cloud provider is used for hosting Kubernetes, the access from pods to the cloud
+metadata API `169.254.169.254` should also be restricted or blocked if not needed
+because it may leak information.
+
+## Pod security
+
+- [ ] RBAC right to `create` workloads is only granted if necessary.
```

Comment:
```suggestion
- [ ] RBAC rights to `create`, `update`, `patch`, `delete` workloads is only granted if necessary.
```
What do you think of adding these additional rights

Reply:
Yes definitely, I wanted to highlight the `create` one because it's especially critical but the other ones are also super important! Would it be relevant to highlight the `create` one?
```suggestion
- [ ] RBAC rights to **`create`**, `update`, `patch`, `delete` workloads is only granted if necessary.
```